dht.go

```diff
@@ -299,18 +299,29 @@ func (dht *IpfsDHT) shouldAddPeerToRoutingTable(c network.Conn) bool {
 	if isRelayAddr(c.RemoteMultiaddr()) {
 		return false
 	}
-	return c.Stat().Direction == network.DirOutbound ||
-		dht.hasSensibleAddressesForPeer(c)
+	if c.Stat().Direction == network.DirOutbound {
+		// we established this connection, so they're definitely diallable.
+		return true
+	}
+	if dht.peerIsOnSameSubnet(c) {
```

Comment:
if we do this here, we don't return false for peers with no addresses at all, or peers with relay addresses.

Reply:
peers with relay addrs are covered are caught in L299. let me look at the 0-addr case.

Reply:
wait no, they're covered iff we're connected to them via a relay, but not if we are directly connected but they advertise relay addrs.